fix: wrong argument type passed when creating foreign key

philly-php opened the pull request:
we were passing a string into the function to create sql to create a foreign key constraint. This function expects an array, since foreign keys can reference multiple columns.

Fixes Issue #13941

Signed-off-by: Leonid Sklyut <lsklyut@kbra.com>

Before submitting pull request, please check that every commit:

- [x ] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [ ] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [ ] Any new functionality is covered by tests

philly-php committed:
fix: wrong argument type passed when creating foreign key

we were passing a string into the function to create sql to create a foreign key constraint. This function expects an array, since foreign keys can reference multiple columns.

Fixes Issue #13941

Signed-off-by: Leonid Sklyut <lsklyut@kbra.com>

nijel reviewed (changes requested):
I think this should be addressed elsewhere, similarly as https://github.com/phpmyadmin/phpmyadmin/pull/14010

philly-php committed:
refactor: cast the foreign field to an array prior to using it

right now, there is no way to select multiple columns for an index, even though mysql allows it. Since the destination_foreign_column will always be a string at the moment, I am casting it to an array prior to using it, as the code expects it to be one.

we were passing a string into the function to create sql to create a foreign key constraint. This function expects an array, since foreign keys can reference multiple columns.

Fixes Issue #13941

Signed-off-by: Leonid Sklyut <lsklyut@kbra.com>

lsklyut commented:
@nijel , I've pulled the array casting up to where we grab the $foreign_field from $destination_foreign_column. As it stands, it does not look like this input would ever be anything but a string, even though the code further down and in _getSQLToCreateForeignKey relies on it being an array. I do think the array logic makes sense, since mysql does allow multi-column foreign keys (although it doesn't look like the UI allows you to do that yet)

lsklyut commented:
this would also fix #14009
